src/shared_modules/utils/stringHelper.h

```diff
@@ -72,10 +72,13 @@ namespace Utils
         auto pos {data.find(toSearch)};
         const auto ret {std::string::npos != pos};
 
-        while (std::string::npos != pos)
+        if (toSearch.compare(toReplace) != 0)
```

Comment:
Good find! But we are going to refactor this method here
- https://github.com/wazuh/wazuh/issues/25547

Lets revert this